Library/Homebrew/cmd/install.rb

```diff
@@ -258,6 +258,7 @@ def install
 
     formulae.each do |f|
       Migrator.migrate_if_needed(f)
+      forbidden_license_check(f) unless ENV["HOMEBREW_FORBIDDEN_LICENSES"].blank?
```

Comment:
Probably want this in `formula_installer.rb` somewhere so it's handled by `upgrade` and `reinstall`, too. Either `def prelude` or `def check_install_sanity`?